qiita_pet/uimodules/prep_template_tab.py

```diff
@@ -35,13 +35,18 @@
 
 
 def _get_accessible_raw_data(user):
-    """Retrieves a tuple of raw_data_id and the last study title for that
+    """Retrieves a tuple of raw_data_id and one study title for that
     raw_data
     """
     d = {}
-    for sid in user.user_studies:
-        for rdid in Study(sid).raw_data():
-            d[int(rdid)] = Study(RawData(rdid).studies[-1]).title
+    accessible_studies = user.user_studies.union(user.shared_studies)
+    for sid in accessible_studies:
+        study = Study(sid)
+        study_title = study.title
+        for rdid in study.raw_data():
+            rdid = int(rdid)
+            if rdid not in d:
```

Comment:
You can actually defaultdict this now, since it is technically part of whatever study you have it in. If the raw data ID gets over-ridden it is still part of the study that overrides the existing one. Also removes the if statement.